Sources/PointFree/Account/AccountMiddleware.swift

```diff
@@ -1,43 +1,47 @@
+import Dependencies
 import Foundation
 import HttpPipeline
 import Models
+import PointFreeDependencies
 import PointFreePrelude
 import PointFreeRouter
 import Prelude
 import Stripe
 import Tuple
 
 func accountMiddleware(
-  conn: Conn<StatusLineOpen, Tuple4<Models.Subscription?, User?, SubscriberState, Account>>
+  conn: Conn<StatusLineOpen, Account>
 )
   -> IO<Conn<ResponseEnded, Data>>
 {
+  @Dependency(\.currentUser) var currentUser
+  @Dependency(\.subscriberState) var subscriberState
 
-  let (_, user, subscriberState, account) = lower(conn.data)
+  let account = conn.data
 
   switch account {
   case let .confirmEmailChange(payload):
     return conn.map(const(payload))
       |> confirmEmailChangeMiddleware
 
   case .index:
-    return conn.map(const(user .*. subscriberState .*. unit))
+    return conn.map(const(currentUser .*. subscriberState .*. unit))
```

Comment:
There are still a few spots I thread the current user through explicitly, such as when need to require a present `User` and redirect when `nil`, although I did update some spots.

As we push more to get rid of `IO` and `EitherIO` I'm wondering if there could be a throwing computed property `loggedInUser: User`, and depending on what error is throw we could possibly catch the error higher up and perform the login+redirect there. Something to discuss soon.